rcl_yaml_param_parser/src/parse.c

```diff
@@ -534,7 +534,7 @@ _validate_name(const char * name, rcutils_allocator_t allocator)
       }
     } else {
       do {
-        size_t len = separator_pos - absolute_namespace - i;
+        size_t len = ((size_t) (separator_pos - absolute_namespace)) - i;
```

Comment:
I know this code isn't new, but this case is a little scary.  In particular, are we sure that `(separator_pos - absolute_namespace)` is always >= `i`?  If not, we can get into a situation where we pass a very large number into `rcutils_strndup`, leading to bad behavior.  Would you mind looking into this a bit and seeing what you can find out?

Reply:
Sure, I'll look into it.

Reply:
I believe that `separator_pos - absolute_namespace` is always >= `i`. 

Let `s` and `a` be `separator_pos`, `absolute_namespace`, respectively. The arrows (`<--`) mean assignment (e.g., `x <-- x + 1` would be `x += 1`). What I see occuring in the do-while loop is as follows:
```
len <-- s - a - i
i <-- i + len = i + x - y - i = x - y
```
Thus in the next iteration

Where `x` >= `y` because `separator_pos` is always >= `absolute_namespace`. That is because `separator_pos` is a pointer to a location in the `char * absolute_namespace`. Note, that on each loop, if the separation character (`/`) doesn't exist, the loop exits.

I tested out a simplified version of what's going on here: https://repl.it/@audrow/Test-validating-namespace.

Reply:
That works for me.  Thanks for looking into it.